org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/server/Server.java

```diff
@@ -149,7 +205,15 @@ public String getSonarLintEngineState() {
       case NEED_UPDATE:
         return NEED_UPDATE;
       case UPDATED:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 9 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)